### DISCUSSION TITLE
Do you have a similar solution for synapse pipelines?

### DISCUSSION THREAD

**Discussion by NowinskiK** [Azure-Player/azure.datafactory.tools#280] [Q&A] [answered]
**Do you have a similar solution for synapse pipelines?**

Yes & no.
Keep reading below.

**Comment by NowinskiK** [comment, marked as answer]
Yes & no.
**Yes**, because I created Synapse edition of the PowerShell module: [azure.synapse.tools](https://github.com/Azure-Player/azure.synapse.tools)
**No**, because I do NOT recommend to use it for Production environment. There are several reasons for that:
1) The PS module is in Public Preview, and probably will never be supported as #adftools is;
2) PowerShell module from Microsoft has many gaps and inconsistencies;
3) This time Microsoft Team developed its own great tool to deploy Synapse artifacts: "Synapse workspace deployment".

Therefore, for Synapse Analytics I do recommend to use ["Synapse workspace deployment"](https://marketplace.visualstudio.com/items?itemName=AzureSynapseWorkspace.synapsecicd-deploy) (task extension to Azure DevOps) to deploy Synapse objects.

**Reply by Alexdc-12** [reply]
I see your recommendation to use the DevOps plugin, but that doesn't support incremental deployment, we really need a solution for incremental deployment,do you have any ideas on how we can achieve that? 

Could your synapse tools be used to feed a list of items that we want to deploy if we used a git diff command to work out what files we want to send?

**Reply by NowinskiK** [reply]
Could you elaborate why you need "incremental deployment" so much and why Microsoft's DevOps extension doesn't fulfil your needs?
> Could your synapse tools be used to feed a list of items that we want to deploy if we used a git diff command to work out what files we want to send?

Potentially yes when you use "Include" list of objects to be deployed. 
Rather than doing git diff (which shouldn't be use in release process) you can copy new method I developed in #adftools :
[Incremental Deployment](https://github.com/Azure-Player/azure.datafactory.tools#incremental-deployment)

The below code is only part of mechanism, but you'll find other parts too:
https://github.com/Azure-Player/azure.datafactory.tools/blob/6844dee2f81264cb1e9a345970c42e5ef890299f/public/Publish-AdfV2FromJson.ps1#L226-L243

**Reply by Alexdc-12** [reply]
The main reason is so that multiple developers can work and put changes into test/production without dragging others work in with it. 
The DevOps plugin only works on the workspace_publish branch which creates a single JSON file with all workspace content, it doesn't yet support incremental deployment. 
Thanks for your comments I will take a look into this tomorrow.
